cypress/e2e/signUp.cy.js

```diff
@@ -1,12 +1,48 @@
 /// <reference types='cypress' />
 /// <reference types='../support' />
 
+import SignUpPageObject from '../support/pages/signupPageObject';
+import HomePageObject from '../support/pages/home.pageObject';
+
+const signUpPage = new SignUpPageObject();
+const homePage = new HomePageObject();
+
 describe('Sign Up page', () => {
+  let user;
+  let newUser;
+
   before(() => {
 
   });
 
-  it('should ...', () => {
+  beforeEach(() => {
+    cy.task('db:clear');
+    cy.task('generateUser').then((generatedUser) => {
+      user = generatedUser;
+    });
+    cy.task('generateUser').then((generatedUser) => {
+      newUser = generatedUser;
+    });
+    signUpPage.visit();
+  });
+
+  it('should provide an ability to sign up with valid credentials', () => {
+    signUpPage.typeUsername(user.username);
+    signUpPage.typeEmail(user.email);
+    signUpPage.typePassword(user.password);
+    signUpPage.clickSignUpBtn();
+    signUpPage.assertSuccessfulMessage();
+
+    homePage.assertHeaderContainUsername(user.username);
+  });
 
+  it('should not provide an ability to sign up with existing email', () => {
+    cy.register(user.email, user.username, user.password);
```

Comment:
Registering the user in the test case might not be necessary if the user is already generated in the `beforeEach` hook. Consider whether this operation is needed for each test or if it can be optimized.